skynet/dgpu/compute.py

```diff
@@ -81,153 +69,140 @@ def prepare_params_for_diffuse(
+@cm
+def maybe_load_model(name: str, mode: str):
+    if mode == 'diffuse':
+        mode = 'txt2img'
 
-    '''
-    def __init__(self, config: dict):
-        self.cache_dir = None
-        if 'hf_home' in config:
-            self.cache_dir = config['hf_home']
+    global _model_name, _model_mode, _model
+    config = load_skynet_toml().dgpu
 
-        self._model_name: str = ''
-        self._model_mode: str = ''
+    if _model_name != name or _model_mode != mode:
+        # unload model
+        _model = None
+        gc.collect()
+        torch.cuda.empty_cache()
 
-        # self.load_model(DEFAULT_INITAL_MODEL, 'txt2img')
+        _model_name = _model_mode = ''
 
-    def log_debug_info(self):
-        logging.info('memory summary:')
-        logging.info('\n' + torch.cuda.memory_summary())
+        # load model
+        if mode == 'upscale':
+            _model = init_upscaler()
 
-    def is_model_loaded(self, name: str, mode: str):
-        if (name == self._model_name and
-            mode == self._model_mode):
-            return True
+        else:
+            _model = pipeline_for(
+                name, mode, cache_dir=config.hf_home)
 
-        return False
+        _model_name = name
+        _model_mode = mode
 
-    def unload_model(self) -> None:
-        if getattr(self, '_model', None):
-            del self._model
+        logging.debug('memory summary:')
+        logging.debug('\n' + torch.cuda.memory_summary())
 
-        gc.collect()
-        torch.cuda.empty_cache()
+    yield _model
+
+
+def compute_one(
+    model,
+    request_id: int,
+    method: str,
+    params: dict,
+    inputs: list[bytes] = [],
+    should_cancel = None
+):
+    total_steps = params['step'] if 'step' in params else 1
+    def inference_step_wakeup(*args, **kwargs):
+        '''This is a callback function that gets invoked every inference step,
+        we need to raise an exception here if we need to cancel work
+        '''
+        step = args[0]
+        # compat with callback_on_step_end
+        if not isinstance(step, int):
+            step = args[1]
+
+        maybe_update_tui(lambda tui: tui.set_progress(step, done=total_steps))
+
+        should_raise = False
+        if should_cancel:
+            should_raise = trio.from_thread.run(should_cancel, request_id)
+
+        if should_raise:
+            logging.warning(f'CANCELLING work at step {step}')
+            raise DGPUInferenceCancelled('network cancel')
+
+        return {}
+
+    maybe_update_tui(lambda tui: tui.set_status(f'Request #{request_id}'))
+
+    inference_step_wakeup(0)
+
+    output_type = 'png'
+    if 'output_type' in params:
+        output_type = params['output_type']
+
+    output = None
+    output_hash = None
+    try:
+        name = params['model']
+
+        match method:
+            case 'diffuse' | 'txt2img' | 'img2img' | 'inpaint':
+                arguments = prepare_params_for_diffuse(
```

Comment:
why are we allowing `diffuse` to be passed in as a method?

Reply:
`diffuse` method is the old way to refer to txt2img requests, in order to have compat with both we handle it for now, but yeah idea is remove it!